contrib/labstack/echo.v4/echotrace.go

```diff
@@ -57,12 +59,35 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 			// serve the request to the next middleware
 			err := next(c)
 			if err != nil {
-				span.SetTag(ext.Error, err)
-				// invokes the registered HTTP error handler
-				c.Error(err)
+				// It is impossible to determine what the final status code of a request is in echo.
+				// This is the best we can do.
+				switch err := err.(type) {
+				case *echo.HTTPError:
+					if cfg.isStatusError(err.Code) {
+						// mark 5xx server error
+						span.SetTag(ext.Error, err)
+					}
+					span.SetTag(ext.HTTPCode, strconv.Itoa(err.Code))
+				default:
+					// Any non-HTTPError errors appear as 5xx errors.
```

Comment:
This comment confused me at first, I think it belongs 3 lines down (above line 76), with a small edit
```suggestion
					// Any non-echo.HTTPError errors appear as a 5xx status code.
```
Maybe also worth explaining why...

Reply:
Actually, it does belong here, but I'll clarify the meaning.
Basically all errors that fall into this `default` section are treated as if they are `500 Server Error`s, both with the code and with the `isStatusError` call.